platforms/core-runtime/files/src/main/java/org/gradle/api/internal/file/archive/impl/FileZipInput.java

```diff
@@ -25,13 +25,28 @@
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
+import java.io.FilterInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.lang.reflect.Field;
 import java.util.Enumeration;
 import java.util.Iterator;
+import java.util.zip.InflaterInputStream;
 import java.util.zip.ZipFile;
 
 public class FileZipInput implements ZipInput {
+    private static final Field FILTER_INPUT_STREAM_IN;
+
+    static {
+        Field filterInputStreamIn;
+        try {
+            filterInputStreamIn = FilterInputStream.class.getDeclaredField("in");
+            filterInputStreamIn.setAccessible(true);
+        } catch (Exception e) {
+            filterInputStreamIn = null;
+        }
+        FILTER_INPUT_STREAM_IN = filterInputStreamIn;
+    }
```

Comment:
🤔 This looks fragile. If we are to keep this, it should be well documented, both why this is needed and how it works. We should also retain the exception in the debug logs; at least the message, if not the stacktrace.